sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -273,3 +228,75 @@ export class RequestResponseLink implements ReqResLink {
     return new RequestResponseLink(session, sender, receiver);
   }
 }
+
+type NormalizedInfo = {
+  statusCode: number;
+  statusDescription: string;
+  errorCondition: string;
+};
+
+// Handle different variations of property names in responses emitted by EventHubs and ServiceBus.
+const getCodeDescriptionAndError = (props: any): NormalizedInfo => {
+  if (!props) props = {};
+  return {
+    statusCode: (props[Constants.statusCode] || props.statusCode) as number,
```

Comment:
At this point we're basically just casting 'any' so we should check the final value and make they weren't both undefined.

Reply:
This wasn't a new addition, I just moved it around.
However, I see your point. Added an if-check to throw an error if the statusCode is undefined in `onMessageReceived` handler.